src/core/trie.js

```diff
@@ -0,0 +1,79 @@
+// @flow
+
+import {type AddressModule} from "./address";
+import {
+  NodeAddress,
+  type NodeAddressT,
+  EdgeAddress,
+  type EdgeAddressT,
+} from "./graph";
+import * as NullUtil from "../util/null";
+
+type Entry<T> = {map: RecursiveMap<T>, value: ?T};
```

Comment:
Looks like this can be exact, and `map` can be read-only?

Reply:
Done